base/shell/explorer/startmnucust.cpp

```diff
@@ -66,17 +66,228 @@ VOID OnClearRecentItems()
+                             NULL,
+                             &dwType,
+                             (LPBYTE) &chBuffer,
+                             &dwSize) == ERROR_SUCCESS)
+        {
+            if (dwType == REG_SZ)
+            {
+                BOOL stat = (lstrcmpiW(chBuffer, L"YES") == 0) ? TRUE : FALSE;
+                if (stat)
+                    dwStatus |= (1 << iItem);
+                else
+                    dwStatus &= ~(1 << iItem);
+            }
+            else if (dwType == REG_DWORD && dwSize == sizeof(REG_DWORD))
+            {
+                DWORD regValue = *(reinterpret_cast<DWORD *>(chBuffer));
+                if (regValue)
+                    dwStatus |= (1 << iItem);
+                else
+                    dwStatus &= ~(1 << iItem);
+            }
+        }
+    }
+    *dwLength = iItem;
+    //  Save Current User options
+    ClassicOption[iItem].IsDirty = dwStatus;
+    RegCloseKey(hKey);
+    return dwStatus;
+}
+
+DWORD UpLoadUserConfData(DWORD &dwUserData)
+{
+    HKEY hKey;
+    DWORD iItem = 0;
+    DWORD nItems = 0;
+    DWORD dwRes;
+
+    while (ClassicOption[nItems].OptionName != NULL)
+        nItems++;
+
+   if (RegCreateKeyExW(HKEY_CURRENT_USER,
+                       L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Advanced",
+                       0,
+                       NULL,
+                       REG_OPTION_NON_VOLATILE,
+                       KEY_SET_VALUE,
+                       NULL,
+                       &hKey,
+                       &dwRes) == ERROR_SUCCESS)
+    {
+        DWORD dwQueryData = ClassicOption[nItems].IsDirty;
+
+        while (iItem < nItems)
+        {
+            if (((dwQueryData >> iItem) & 0x01) != ((dwUserData >> iItem) & 0x01))
+            {
+                if (((IDS_DRAG_DROP + iItem) > IDS_SHOW_RUN) &&
+                    ((IDS_DRAG_DROP + iItem) < IDS_SHOW_SMALL_ICO) &&
+                    ((IDS_DRAG_DROP + iItem) != IDS_SHOW_FAVORITES))
+                {
+                    const wchar_t *sKey = ((dwQueryData >> iItem) & 0x01) ? L"YES" : L"NO";
+                    RegSetValueExW(hKey, ClassicOption[iItem].OptionName, NULL, REG_SZ, (LPBYTE) sKey, (lstrlen(sKey) + 1) * sizeof(WCHAR));
+                }
+                else
+                {
+                    DWORD regValue = ((dwQueryData >> iItem) & 0x01) ? 1 : 0;
+                    RegSetValueExW(hKey, ClassicOption[iItem].OptionName, NULL, REG_DWORD, (LPBYTE) &regValue, sizeof(REG_DWORD));
+                }
+            }
+            iItem++;
+        }
+        return dwQueryData;
+    }
+    return 0;
+}
+
+HWND InitClassicListView(HWND hwnd)
+{
+    RECT rcClient;
+    HWND hListView = GetDlgItem(hwnd, IDC_CLASSICSTART_SETTINGS);
+    ListView_SetExtendedListViewStyle(hListView, LVS_EX_CHECKBOXES | LVS_EX_FULLROWSELECT);
+    GetClientRect(hListView, &rcClient);
+    WCHAR chBuffer[80];
+    int id = IDS_DRAG_DROP;
+    int nItems = 0;
```

Comment:
I don't enjoy reading C++ code where code and variable definitions are so mixed together that it looks like a brick of letters...

Reply:
You right. Me too. Thanks